### PR TITLE
Get setPath() to work once again.

### PR DESCRIPTION
**Pull request by dwrensha**
Also, openSession was referencing some non-existent fields like `self._query`.

### CODE REVIEW

**Comment by dwrensha** [comment]
Garply, retest this please

**Comment by dwrensha** [comment]
To save you a click-through to Jenkins, this is the error from the first build attempt:

```
[FAILED ] compile: libseccomp/arch.c
/ekam-provider/canonical/libseccomp/arch.c:43:10: fatal error: 'arch-s390.h' file not found
#include "arch-s390.h"
         ^
1 error generated.
[FAILED ] compile: libseccomp/arch-syscall-dump.c
/ekam-provider/canonical/libseccomp/arch-syscall-dump.c:43:10: fatal error: 'arch-s390.h' file not found
#include "arch-s390.h"
         ^
1 error generated.
```

File: shell/client/grainview.js [hunk 1]
```diff
@@ -401,9 +410,9 @@ GrainView.prototype.sessionStatus = function () {
 GrainView.prototype.route = function () {
   this._dep.depend();
   if (this._token) {
-    return "/shared/" + this._token;
+    return "/shared/" + this._token + this._path;
```
Comment (zarvox) [inline shell/client/grainview.js]: We should probably also include the query string and fragment in both of these too?

Comment (dwrensha) [inline shell/client/grainview.js]: Hm. We could include the original query and the original fragment. But what if someone calls setPath() with a new query and a new fragment? I don't think we've nailed down exactly the semantics we want there. Should we ignore any query and fragment that comes in on setPath()?

Comment (dwrensha) [inline shell/client/grainview.js]: It seems clear that we do want `setPath()` to be able to change the query and fragment. I guess the question is: should they be passed as separate arguments, or should they be part of the single `path` argument? I think we want separate arguments, but in any case it's a new feature and I think it should be a separate pull request.

**Comment by dwrensha** [comment]
Garply, retest this please

File: shell/client/grainview.js (hunk 1 above)
Comment (kentonv) [inline shell/client/grainview.js]: I think we should probably keep it as just a string that the app provides. Attempting to parse the query could break some apps that do stupid things like use the same query parameter name multiple times, and I don't think it gains us anything.

We my want to consider what happens if the app sends "../foo" as the path... but not now.

Comment (dwrensha) [inline shell/client/grainview.js]: Oops, I just realized that as this is now, the query and fragment never get through at all, even on the first run.
